api/python/quilt3/api.py

```diff
@@ -151,6 +165,14 @@ def list_package_versions(name, registry=None):
     Returns:
         A sequence of tuples containing the named version and hash.
     """
+    return _list_package_versions(name=name, registry=registry)
+
+
+def _list_package_versions(name, registry=None):
```

Comment:
the `_method` = telemetry-free makes sense but, just a thought, is there an easier way to do this that doesn't require method renaming? idk like  global singleton where you can turn off telemetery; i don't love that either just wondering aloud if there's another pattern for this already.

Reply:
I was thinking maybe we should move to a design where we have an `InternalPackage` class that hold the internal logic and a `Package` class that is for the end-user. We put telemetry and documentation on the `Package` class methods which ensures we never get more than one event per user action. It would also clarify what APIs we can change freely and what APIs we are promising the user will stay the same. 

But that is a bit more of a substantial change.